lib/cisco_node_utils/fabricpath_global.rb

```diff
@@ -0,0 +1,439 @@
+
+module Cisco
+  # node_utils class for fabricpath_global
+  class FabricpathGlobal < NodeUtil
+    attr_reader :name
+
+    def initialize(name, instantiate=true)
+      fail TypeError unless name.is_a?(String)
+      fail ArgumentError unless name == 'default'
+      @name = name.downcase
+      @set_params = {}
+
+      create if instantiate
+    end
+
+    def self.globals
+      hash = {}
+      is_fabricpath_feature = config_get('fabricpath', 'feature')
+      return hash if (:enabled != is_fabricpath_feature.to_sym)
+      hash['default'] = FabricpathGlobal.new('default', false)
+      hash
+    end
+
+    def self.fabricpath_feature
+      fabricpath = config_get('fabricpath', 'feature')
+      fail 'fabricpath_feature not found' if fabricpath.nil?
+      return :disabled if fabricpath.nil?
+      fabricpath.to_sym
+    end
+
+    def self.fabricpath_feature_set(fabricpath_set)
+      curr = FabricpathGlobal.fabricpath_feature
+      return if curr == fabricpath_set
+
+      case fabricpath_set
+      when :enabled
+        config_set('fabricpath', 'feature_install',
+                   state: '') if curr == :uninstalled
+        config_set('fabricpath', 'feature', state: '')
+      when :disabled
+        config_set('fabricpath', 'feature', state: 'no') if curr == :enabled
+        return
+      when :installed
+        config_set('fabricpath', 'feature_install',
+                   state: '') if curr == :uninstalled
+      when :uninstalled
+        config_set('fabricpath', 'feature', state: 'no') if curr == :enabled
+        config_set('fabricpath', 'feature_install', state: 'no')
+      end
+    rescue Cisco::CliError => e
+      raise "[#{@name}] '#{e.command}' : #{e.clierror}"
+    end
+
+    def create
+      FabricpathGlobal.fabricpath_feature_set(:enabled) unless
+      :enabled == FabricpathGlobal.fabricpath_feature
+    end
+
+    def destroy
+      @name = nil
+      FabricpathGlobal.fabricpath_feature_set(:disabled)
+    end
+
+    def my_munge(property, set_val)
+      val = config_get_default('fabricpath', property)
+      case property
+      when /loadbalance_algorithm/
+        if (val == 'source-destination') &&
+           (set_val == 'symmetric' || set_val == 'xor')
+          val
+        else
+          set_val
+        end
+      when /loadbalance_.*_rotate/
+        if val == false || set_val == ''
+          ''
+        else
+          "rotate-amount 0x#{set_val.to_s(16)}"
+        end
+      else
+        set_val
+      end
+    end
+
+    ########################################################
+    #                      PROPERTIES                      #
+    ########################################################
+
+    def aggregate_multicast_routes
+      config_get('fabricpath', 'aggregate_multicast_routes')
+    end
+
+    def aggregate_multicast_routes=(val)
+      @set_params = {}
+      @set_params[:state] = val ? '' : 'no'
+      config_set('fabricpath', 'aggregate_multicast_routes', @set_params)
```

Comment:
There's no need for @set_params since you're only using it for a single key (or two in some cases). Just do:

```
config_set('fabricpath', 'aggregate_multicast_routes', state: val ? '' : 'no')
```